uberfire-extensions/uberfire-metadata/uberfire-metadata-backends/uberfire-metadata-backend-infinispan/src/main/java/org/uberfire/ext/metadata/backend/infinispan/provider/MappingProvider.java

```diff
@@ -0,0 +1,246 @@
+import org.uberfire.ext.metadata.model.impl.KPropertyImpl;
+import org.uberfire.java.nio.base.version.VersionHistory;
+import org.uberfire.java.nio.file.attribute.FileTime;
+
+public class MappingProvider {
+
+    public static final String CHECKIN_COMMENT = "checkinComment";
+    public static final String LAST_MODIFIED_BY = "lastModifiedBy";
+    public static final String CREATED_BY = "createdBy";
+    public static final String CREATED_DATE = "createdDate";
+    public static final String LAST_MODIFIED_DATE = "lastModifiedDate";
+
+    public static final String ORG_KIE = "org.kie";
+    private Analyzer analyzer;
+
+    public MappingProvider(Analyzer analyzer) {
+
+        this.analyzer = analyzer;
+    }
+
+    public Schema getMapping(KObject kObject) {
+        return this.buildSchema(kObject);
+    }
+
+    private Schema buildSchema(KObject kObject) {
+        return new Schema(kObject.getClusterId(),
+                          ORG_KIE,
+                          this.buildMessages(kObject));
+    }
+
+    public Schema buildSchema(KObject kObject,
+                              Set<Message> messages) {
+        return new Schema(kObject.getClusterId(),
+                          ORG_KIE,
+                          messages);
+    }
+
+    public Set<Message> buildMessages(KObject kObject) {
+
+        List<Field> fields = new ArrayList<>();
+        fields.add(this.createField(ProtobufScope.OPTIONAL,
+                                    ProtobufType.STRING,
+                                    "id"));
+        fields.add(this.createField(ProtobufScope.REQUIRED,
+                                    ProtobufType.STRING,
+                                    "type"));
+        fields.add(this.createField(ProtobufScope.REQUIRED,
+                                    ProtobufType.STRING,
+                                    "cluster__id"));
+        fields.add(this.createField(ProtobufScope.REQUIRED,
+                                    ProtobufType.STRING,
+                                    "segment__id"));
+        fields.add(this.createField(ProtobufScope.REQUIRED,
+                                    ProtobufType.STRING,
+                                    "key"));
+        fields.add(createField(ProtobufScope.OPTIONAL,
+                               ProtobufType.STRING,
+                               "fullText",
+                               true,
+                               true));
+
+        Iterator<KProperty<?>> iterator = kObject.getProperties().iterator();
+        while (iterator.hasNext()) {
+            KProperty<?> prop = iterator.next();
+            fields.addAll(this.build(prop));
+        }
+
+        Message message = new Message(sanitize(kObject.getClusterId() + "_" + kObject.getType().getName()),
+                                      new HashSet<>(fields));
+
+        return Collections.singleton(message);
+    }
+
+    private String sanitize(String name) {
```

Comment:
This helper seems useles. Why not just inline it and use `toProtobufFormat` directly?